charmcraft/commands/build.py

```diff
@@ -177,11 +183,7 @@ def run(self, bases_indices: Optional[List[int]] = None) -> List[str]:
         """
         charms: List[str] = []
 
-        if is_charmcraft_running_in_managed_mode():
-            if self.config.bases is None:
-                # XXX: 2021-06-16 Patterson temporary until we have the default base
-                raise CommandError("Bases are currently required in managed-mode.")
-
+        if self.config.bases:
             for i, bases_config in enumerate(self.config.bases):
```

Comment:
given the size of this, can we separate selection/planning from execution?

Reply:
Sounds good, will do!